iroh/src/client.rs

```diff
@@ -104,37 +178,99 @@ where
         };
         Ok(Some(doc))
     }
+}
+
+/// Iroh blobs client.
+#[derive(Debug, Clone)]
+pub struct BlobsClient<C> {
+    rpc: RpcClient<ProviderService, C>,
+}
+
+impl<C> BlobsClient<C>
+where
+    C: ServiceConnection<ProviderService>,
+{
+    /// Read a single blob as a streaming [`BlobReader`].
+    pub async fn read(&self, hash: Hash) -> Result<BlobReader> {
+        BlobReader::from_rpc(&self.rpc, hash).await
+    }
 
-    /// Get the bytes for a hash.
+    /// Read all bytes of single blob.
     ///
-    /// Note: This reads the full blob into memory.
-    pub async fn get_bytes(&self, hash: Hash) -> Result<Bytes> {
-        let mut stream = self.get_bytes_stream(hash).await?;
-        stream.read_to_end().await
+    /// This allocates a buffer for the full blob. Use only if you know that the blob you're
+    /// reading is small. If not sure, use [`Self::read`] and check the size with
+    /// [`BlobReader::size`] before calling [`BlobReader::read_to_end`].
+    pub async fn read_to_end(&self, hash: Hash) -> Result<Bytes> {
+        BlobReader::from_rpc(&self.rpc, hash)
+            .await?
+            .read_to_end()
+            .await
     }
 
-    /// Get the bytes for a hash.
-    pub async fn get_bytes_stream(&self, hash: Hash) -> Result<BlobReader> {
-        BlobReader::from_rpc(&self.rpc, hash).await
+    /// Import a blob from a filesystem path.
+    ///
+    /// `path` should be an absolute path valid for the file system on which
+    /// the node runs.
+    /// If `in_place` is true, Iroh will assume that the data will not change and will share it in
+    /// place without copying to the Iroh data directory.
+    pub async fn add_from_path(
+        &self,
+        path: PathBuf,
+        in_place: bool,
+    ) -> Result<impl Stream<Item = Result<AddProgress>>> {
+        let stream = self
+            .rpc
+            .server_streaming(BlobAddPathRequest { path, in_place })
+            .await?;
+        Ok(stream.map_err(anyhow::Error::from))
     }
 
-    /// Get statistics of the running node.
-    pub async fn stats(&self) -> Result<HashMap<String, CounterStats>> {
-        let res = self.rpc.rpc(StatsGetRequest {}).await??;
-        Ok(res.stats)
+    /// Validate hashes on the running node.
+    ///
+    /// If `repair` is true, repair the store by removing invalid data.
+    pub async fn validate(
```

Comment:
shouldn't this be on the `node` part of things?

Reply:
Not sure - it deals with blobs only. No strong opinion.